ethcore/src/client/client.rs

```diff
@@ -969,19 +977,26 @@ impl Client {
 
 	/// Import transactions from the IO queue
 	pub fn import_queued_transactions(&self, transactions: &[Bytes], peer_id: usize) -> usize {
-		trace!(target: "external_tx", "Importing queued");
 		trace_time!("import_queued_transactions");
 		self.queue_transactions.fetch_sub(transactions.len(), AtomicOrdering::SeqCst);
-		let txs: Vec<UnverifiedTransaction> = transactions.iter().filter_map(|bytes| UntrustedRlp::new(bytes).as_val().ok()).collect();
+
+		let txs: Vec<UnverifiedTransaction> = transactions
+			.iter()
+			.filter_map(|bytes| UntrustedRlp::new(bytes).as_val().ok())
+			.collect();
+
+		// Notify sync that the transactions were received from given peer
 		let hashes: Vec<_> = txs.iter().map(|tx| tx.hash()).collect();
```

Comment:
sure, although I would classify it as beyond the scope of the PR. this PR only changes behavior of the underlying queue and only changes code around it to the extent that is needed. I think it is large enough already and I've filed an issue at #8082

Reply:
Unfortunately, I was wrong that the change is trivial. Both `txs` and `hashes` are actually used in the code below. 

However, at least, we may eliminate `hashes.clone()` by changing `transactions_received` callback to accept  `&[H256]`, `Cow` or `Rc`, instead of bare `Vec<H256>`. Not every handler would like to own that vector. But current implementation is forced to clone `hashes` to pass it to handler even it has nothing to do with it, like default one.